maggy/core/experimentdriver.py

```diff
@@ -109,6 +110,10 @@ def __init__(self, experiment_type, **kwargs):
                     self.optimizer = RandomSearch()
                 elif optimizer.lower() == "asha":
                     self.optimizer = Asha()
+                elif optimizer.lower() == "tpe":
+                    self.optimizer = bayes.TPE()
+                elif optimizer.lower() == "gp":
+                    self.optimizer = bayes.GP()
```

Comment:
at this point pruner and optimizer are initialized, right? can we call the `pruner.num_trials()` method here to set it in the experiment driver for the progress bar? we do the same in line 209 for ablation studies.

Reply:
yes

Reply:
we need to additionally add if/else clause in line 270, otherwise it gets overwritten again